utilities/transactions/transaction_lock_mgr.cc

```diff
@@ -414,15 +421,29 @@ bool TransactionLockMgr::IncrementWaiters(
       return false;
     }
 
-    auto next = queue[head];
+    auto next = queue[head].first;
     if (next == id) {
+      std::vector<DeadlockInfo> path;
+      while (head != -1) {
+        uint32_t tracked_cf_id = 0;
+        std::string tracked_waiting_key = "";
+        if (wait_txn_map_.Contains(queue[head].first)) {
```

Comment:
Is it ever valid that wait_txn_map would be missing what you're looking for? Should we be asserting instead?